lib/spack/spack/compilers/nag.py

```diff
@@ -20,6 +20,10 @@ class Nag(Compiler):
                    'f77' : 'nag/nagfor',
                    'fc'  : 'nag/nagfor' }
 
+    # Needs double -Wl for rpaths.
+    f77_rpath_arg = '-Wl,-Wl,-rpath,%s'
```

Comment:
You're missing a comma. It needs to be `'-Wl,-Wl,,-rpath,%s'`. Otherwise I still get the `gcc: unrecognized command line option '-rpath'` error message.